src/core/geometry/qgsgeometryutils.cpp

```diff
@@ -1810,6 +1806,32 @@ void QgsGeometryUtils::weightedPointInTriangle( const double aX, const double aY
   pointY = rBy + rCy + aY;
 }
 
+bool QgsGeometryUtils::transferFirstZOrMValueToPoint( const QgsPointSequence &points, QgsPoint &point )
+{
+  bool z_passed = false;
+  bool m_passed = false;
+
+  for ( const QgsPoint &pt : points )
+  {
+    if ( !m_passed && pt.isMeasure() )
+    {
+      point.convertTo( QgsWkbTypes::addM( point.wkbType() ) );
+      point.setM( pt.m() );
```

Comment:
```suggestion
      point.addMValue( pt.m() );
```

Reply:
This is what I wanted to do at the beginning (see my comment in https://github.com/qgis/QGIS/pull/42812). 
Currently, the method adds the Z (or M) from the list of points no matter if it already exists. 
By using addZValue (or M), we write only if the point has no Z (or M) yet and it is therefore no longer consistent with the description of the function (and its name).

Reply:
ok, fair enough!